### ISSUE TITLE
question / feature request: allow recording in full color range

### ISSUE THREAD

**Issue by tari3x** [tonioni/WinUAE#264]
Currently the color range of the recordings seems to be limited:
```
ffprobe -f lavfi movie="capture.avi",signalstats -show_entries frame_tags=lavfi.signalstats.YMAX,lavfi.signalstats.YMIN
...
[FRAME]
TAG:lavfi.signalstats.YMIN=16
TAG:lavfi.signalstats.YMAX=235
[/FRAME]
```

I suspect this is expected since a real amiga would output to a TV set which has limited range. But just in case I'm confused here's a feature request - please allow to record in full color range. 

Afaik utvideo + avi don't have anywhere to mark the video as full range, so it would need to be mkv.

**Comment by luvwagn** [comment]
Could also be done into an MP4 container, with AVC/H264 as the encoding type?

**Comment by tonioni** [comment]
AVI recording was done by someone else long, long, long time ago and I am not interested in rewriting it to use some modern method (video recording is probably my lowest priority feature ever..). Perhaps if there is some really good license compatible open source example somewhere.

**Comment by tari3x** [comment]
> video recording is probably my lowest priority feature ever..

That's sad actually - I suspect recording amiga demos is a rather sizeable use case.

**Comment by tonioni** [comment]
I don't think it is that simple, for example does it support uncompressed? Thats very nice feature of AVI stuff, you can select uncompressed (can easily test if recording is exactly right) and IMHO it is better to record it in some non-lossy format (AVI has non-lossy formats which someone really wants to have), edit the file and then convert it to whatever format you want.

Anyway, as I said, I need example sources that is proven working, that includes CODEC selection GUI. I am not interested in reading some API docs that usually always suck and assume you already know what are you doing :)
(It does not need to be emulator, just some program that records something)

**Comment by johnnovak** [comment]
> I don't think it is that simple, for example does it support uncompressed? Thats very nice feature of AVI stuff, you can select uncompressed (can easily test if recording is exactly right) and IMHO it is better to record it in some non-lossy format (AVI has non-lossy formats which someone really wants to have), edit the file and then convert it to whatever format you want.
> 
> Anyway, as I said, I need example sources that is proven working, that includes CODEC selection GUI. I am not interested in reading some API docs that usually always suck and assume you already know what are you doing :) (It does not need to be emulator, just some program that records something)

Video encoding is a huge rabbit hole agreed. We're experiencing that in DOSBox Staging now 😅 Lossless is the best option, then people can encode that however they like 🤷🏻 ...unless you want to mess around with vendor-specific APIs to add hardware accelerated encoding support, which is nice, but fiddly.

**Comment by luvwagn** [comment]
What formats can I get the frames in as input - D3D textures with a clear DXGI format?  raw RGB data per frame?  Let me know which works best, and I'll happily code up an e2e example and share it with you.  I'm a part owner of these technologies in Windows.
________________________________
